samples/BankAccount/AccountTransfer.Grains/AccountGrain.cs

```diff
@@ -1,41 +1,44 @@
 using AccountTransfer.Interfaces;
-using Orleans;
+using Orleans.Concurrency;
 using Orleans.Transactions.Abstractions;
 
 namespace AccountTransfer.Grains;
 
-[Serializable]
+[GenerateSerializer]
 public record class Balance
 {
-    public uint Value { get; set; } = 1_000;
+    [Id(0)]
+    public int Value { get; set; } = 1_000;
 }
 
-public class AccountGrain : Grain, IAccountGrain
+[Reentrant]
+public sealed class AccountGrain : Grain, IAccountGrain
 {
     private readonly ITransactionalState<Balance> _balance;
 
     public AccountGrain(
         [TransactionalState("balance")] ITransactionalState<Balance> balance) =>
         _balance = balance ?? throw new ArgumentNullException(nameof(balance));
 
-    public Task Deposit(uint amount) =>
+    public Task Deposit(int amount) =>
         _balance.PerformUpdate(
             balance => balance.Value += amount);
 
-    public Task Withdraw(uint amount) =>
+    public Task Withdraw(int amount) =>
         _balance.PerformUpdate(balance =>
         {
             if (balance.Value < amount)
             {
-                throw new InvalidOperationException(
-                    $"Withdrawing {amount} credits from account " +
-                    $"\"{this.GetPrimaryKeyString()}\" would overdraw it." +
-                    $" This account has {balance.Value} credits.");
+                throw new InvalidOperationException($"""
+                    Withdrawing {amount} credits from account 
+                    "{this.GetPrimaryKeyString()}" would overdraw it.
```

Comment:
This will insert newlines into the exception, which wasn't the intention in the original

Reply:
Good catch, I'll revert that.